contracts/base/Dispatcher.sol

```diff
@@ -316,11 +315,22 @@ abstract contract Dispatcher is Payments, V2SwapRouter, V3SwapRouter, Callbacks
             }
             // 0x20 <= command
         } else {
-            // placeholder area for commands 0x20-0x3f
-            revert InvalidCommandType(command);
+            if (command == Commands.EXECUTE_SUB_PLAN) {
+                (bytes memory _commands, bytes[] memory _inputs) = abi.decode(inputs, (bytes, bytes[]));
+                (success, output) =
+                    (address(this)).call(abi.encodeWithSignature('execute(bytes,bytes[])', _commands, _inputs));
+            } else {
+                // placeholder area for commands 0x21-0x3f
+                revert InvalidCommandType(command);
+            }
         }
     }
 
+    /// @notice Executes encoded commands along with provided inputs.
+    /// @param commands A set of concatenated commands, each 1 byte in length
+    /// @param inputs An array of byte strings containing abi encoded inputs for each command
+    function execute(bytes calldata commands, bytes[] calldata inputs) external payable virtual;
```

Comment:
kind of awkward. If I was coding it, I'd probably just allow the dispatcher to arbitrarily call back into itself like we call the protocols but maybe being more rigid is the right move despite the awkwardness?

Reply:
This function actually doesnt _have_ to be here because we do the call to `execute` as a low level call. However that low level call does of course mean that all inheriting contracts must implement an execute function - which is why i put this here.